find_number.rb

```diff
@@ -0,0 +1,39 @@
+is_session_active = true
+is_game_active = true
+
+while is_session_active
+  num_to_guess = rand(1..100)
+
+  while is_game_active
+
+    puts "Guess a number between 1 and 100"
+    guess = gets.chomp
+
+    until guess.to_i >= 1 && guess.to_i <= 100
+      puts "Invalid guess. Please guess a number between 1 and 100"
+      guess = gets.chomp
+    end
```

Comment:
Ah, this is clever.

You may also like the [`#between?`](https://ruby-doc.org/3.2.2/Comparable.html#method-i-between-3F) method (note: I am demonstrating as if you accepted my above suggestion to convert `guess` to an integer when declaring it):

```ruby
guess.between?(1,100)
```